lucille-core/src/main/java/com/kmwllc/lucille/core/RunDetails.java

```diff
@@ -0,0 +1,155 @@
+package com.kmwllc.lucille.core;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.kmwllc.lucille.core.Runner.RunType;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@Schema(
+    description = "Details about a specific run, including metadata, start time, and processing stats.")
+public class RunDetails {
+
+  private static final Logger log = LoggerFactory.getLogger(RunDetails.class);
+
+  @Schema(description = "Unique identifier for the run.", example = "run123")
+  private final String runId;
+
+  @Schema(description = "Configuration ID for the run.")
+  private String configId;
+
+  @Schema(description = "Start time of the run in ISO-8601 format.",
+      example = "2023-12-20T14:48:00.000Z")
+  private final Instant startTime;
+
+  @Schema(
+      description = "End time of the run in ISO-8601 format. Null if the run has not completed.",
+      example = "2023-12-20T15:48:00.000Z")
+  private Instant endTime;
+
+  @Schema(description = "Number of documents processed during the run.", example = "5000")
+  private long docsProcessed;
```

Comment:
as commented elsewhere, this duplicates information already stored in RunResult -- please remove

Reply:
I know there was interest in seeing the current running status of documents processed, I don't know where that is or how it can be retrieved.  Removing now.